src/main/include/subsystems/HatchIntake.h

```diff
@@ -0,0 +1,32 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2017-2018 FIRST. All Rights Reserved.                        */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+#ifndef SRC_SUBSYSTEMS_HATCHINTAKE_H_
+#define SRC_SUBSYSTEMS_HATCHINTAKE_H_
+#pragma once
+
+#include <frc/commands/Subsystem.h>
+#include "RobotMap.h"
+#include "frc/DoubleSolenoid.h"
+#include "ctre/Phoenix.h"
+
+class HatchIntake : public frc::Subsystem {
+ private:
+  // It's desirable that everything possible under private except
+  // for methods that implement subsystem capabilities
+  static const char kSubsystemName[];
+  static std::shared_ptr<HatchIntake> self;
+  frc::DoubleSolenoid hatchPuncher;
```

Comment:
`hatchPuncher` is part of `HatchIntake`, `hatchPuncher` is redundant. e.g. `hatchIntake->hatchPuncher`.

Rename `hatchPuncher` to `puncher`.